packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -669,6 +672,54 @@ export function authorizationServiceBuilder(
       }
       return key;
     },
+    async getKeyWithClientByKeyId({
+      clientId,
+      kid,
+      logger,
+    }: {
+      clientId: ClientId;
+      kid: string;
+      logger: Logger;
+    }): Promise<KeyWithClient> {
+      logger.info(`Getting client ${clientId} and key ${kid}`);
+      const client = await retrieveClient(clientId, readModelService);
+      const key = client.data.keys.find((key) => key.kid === kid);
+
+      if (!key) {
+        throw keyNotFound(kid, clientId);
+      }
+
+      const jwk: JsonWebKey = createJWK(decodeBase64ToPem(key.encodedPem));
+      const JwkKey: JWKKey = {
+        kty: jwk.kty!,
```

Comment:
You have to check kty in order to create the proper JWKKey as it's done [here](https://github.com/pagopa/interop-be-authorization-management/blob/f5250fdaac7b7e4f5c1a01bf7739b202b8d48db7/keyconverter/src/main/scala/it/pagopa/interop/authorizationmanagement/jwk/converter/KeyConverter.scala#L75)